AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java

```diff
@@ -428,13 +439,30 @@ private void bindOnVideoStreamsUpdatedListener(final RemoteParticipant remotePar
             if (!displayedRemoteParticipantIds.contains(id)) {
                 return;
             }
+            if (!lookUpNewScreenSharing(remoteParticipant)) {
+                if (currentScreenSharingParticipant != null
+                        && id.equals(getId(currentScreenSharingParticipant))) {
+                    currentScreenSharingParticipant = null;
+                }
+            }
             displayedParticipantsLiveData.postValue(displayedRemoteParticipants);
             Log.d(LOG_TAG, String.format("Remote Participant %s addOnRemoteVideoStreamsUpdatedListener", username));
         };
         remoteParticipant.addOnVideoStreamsUpdatedListener(remoteVideoStreamsUpdatedListener);
         videoStreamsUpdatedListenersMap.put(id, remoteVideoStreamsUpdatedListener);
     }
 
+    private boolean lookUpNewScreenSharing(final RemoteParticipant remoteParticipant) {
```

Comment:
I feel like this function can be improved.  The positive case changes some state, while the negative case returns a boolean which makes a conditional change a state.

Reply:
Addressed and changed the function.